lib/driver.js

```diff
@@ -1187,13 +1332,16 @@ class XCUITestDriver extends BaseDriver {
       maxTypingFrequency: this.opts.maxTypingFrequency ?? 60,
       shouldUseSingletonTestManager: this.opts.shouldUseSingletonTestManager ?? true,
       waitForIdleTimeout: this.opts.waitForIdleTimeout,
+      // @ts-expect-error - do not assign arbitrary properties to `this.opts`
       shouldUseCompactResponses: this.opts.shouldUseCompactResponses,
```

Comment:
the things in this list are capabilities, so they should be OK on `this.opts` (i.e., they're not arbitrary properties). probably they're missing from the cap constraints.

Reply:
hmm, which?

Reply:
ehh maybe just this one. I'll add it

Reply:
@jlipps OK, so I'm looking at this, and `forceSimulatorSoftwareKeyboardPresence` is mentioned in the capability docs.

`shouldUseCompactRepsonses` is mentioned in the _settings_ documentation, as is `elementResponseAttributes` (both in this data structure).

Should these _all_ be added as constraints?  I also note that _`elementResponseFields`_ is referenced several times in the codebase (instead of +`elementResponseAttributes`_), which may be a bug?

cc @mykola-mokhnach @KazuCocoa

Reply:
For `elementResponseFields`. It is maybe no longer used since it is not in appium/WDA as well. It is in unit tests and capabilities for WDA, but no places use the value

Reply:
An item that was addressed only in settings can be set only via `settings[<settings name>]` as capabilities for now, as a documented way. 

Some can be set via capabilities as https://github.com/appium/WebDriverAgent/blob/master/WebDriverAgentLib/Commands/FBSessionCommands.m#L101-L123

Maybe we should print deprecated messages in xcuitest about capabilities that are available as capabilities (as implementation) for WDA, but not in capabilities docs. Then remove them from capabilities completely.